Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,64 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _security_identities:
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint32``,
+which means the maximum limit for a security identity is ``2^32 - 1``. The
+minimum security identity is ``1``, due to the reserved identity range being
+from ``1`` to ``255``. See the `internal code documentation
+<https://pkg.go.dev/github.com/cilium/cilium/pkg/identity#NumericIdentity>`_.
+
+.. note::
+
+   Identitiy 0 is not a valid value. If it shows up in Hubble output, this
+   means the identity was not found. In the eBPF datapath, it has a special
+   role where it denotes "any identity", i.e. as a wildcard allow in policy
+   map.
+
+There are two caveats to the above however:
+
+1) ClusterMesh
+2) Identities generated from CIDR-based policies
+
+For (1), an extra 8 bits are added as the ``cluster-id`` which identifies the
+cluster in the ClusterMesh, into the 3rd octet as shown by ``0x00FF0000``. This
+makes the security identity's upper limit ``2^24 - 1``. This does not apply to
+CIDR identities however, see (2).
+
+For (2), CIDR identities are local to each node. This means that they're not
+subject to the global limit of security identities (``2^16 - 1``). CIDR
+identities begin from ``1`` and end at ``16777215``, however since they're
+shifted by ``24``, this makes their effective range ``1 | (1 << 24)`` to
+``16777215 | (1 << 24)`` or from ``16777217`` to ``33554431``.  The reason that
+CIDR identities are local is because of the limitation imposed by the VXLAN or
+Geneve virtual network field size being 24 bits.
+
+.. note::
```

Comment:
Do we need this to be a dedicated call-out box? I think that it's fine to just drop the note / indent and let the following paragraphs flow on from this, there's no reason to highlight the below information vs. the other information in this doc.

Reply:
^^ If we end up with documentation where half of the writing is in `note::` sections and half isn't, then the way that the documentation UI presents the information to the user will only emphasize the information in the notes and discourage reading the non-note content. In the end this means that rather than structuring the information, we place unintended emphasis on some information.

I'll generally encourage attempting to write all docs without any sort of notes or callouts first, and only then when you have a full document, decide whether some of the information should be emphasized with `note` or `warning` sections.

Reply:
Hm good point, I didn't think of it like that. Good to know for the future :+1: